service/chain/disk_test.go

```diff
@@ -32,286 +30,407 @@ import (
 )
 
 func TestDisk_Root(t *testing.T) {
-	db := populateDB(t)
-	defer db.Close()
-	c := chain.FromDisk(db)
+	t.Run("nominal case", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		require.NoError(t, db.Update(operation.InsertRootHeight(mocks.GenericHeight)))
+
+		c := chain.FromDisk(db)
+
+		root, err := c.Root()
+		assert.NoError(t, err)
+		assert.Equal(t, mocks.GenericHeight, root)
+	})
+
+	t.Run("handles missing root height entry in db", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		c := chain.FromDisk(db)
+
+		_, err := c.Root()
+		assert.Error(t, err)
+	})
 
-	root, err := c.Root()
-	assert.NoError(t, err)
-	assert.Equal(t, mocks.GenericHeight, root)
 }
 
 func TestDisk_Header(t *testing.T) {
-	db := populateDB(t)
-	defer db.Close()
-	c := chain.FromDisk(db)
+	t.Run("nominal case", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		require.NoError(t, db.Update(operation.IndexBlockHeight(mocks.GenericHeight, mocks.GenericHeader.ID())))
+		require.NoError(t, db.Update(operation.InsertHeader(mocks.GenericHeader.ID(), mocks.GenericHeader)))
+
+		c := chain.FromDisk(db)
+
+		header, err := c.Header(mocks.GenericHeight)
+		assert.NoError(t, err)
+
+		require.NotNil(t, header)
+		assert.Equal(t, dps.FlowTestnet, header.ChainID)
 
-	header, err := c.Header(mocks.GenericHeight)
-	assert.NoError(t, err)
+		_, err = c.Header(math.MaxUint64)
+		assert.Error(t, err)
+	})
+
+	t.Run("handles missing entry for indexed height", func(t *testing.T) {
+		t.Parallel()
+
+		// Only index the block height, but no header for that height.
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		require.NoError(t, db.Update(operation.IndexBlockHeight(mocks.GenericHeight, mocks.GenericHeader.ID())))
+
+		c := chain.FromDisk(db)
+
+		_, err := c.Header(mocks.GenericHeight)
+		assert.Error(t, err)
+	})
+
+	t.Run("handles call on non-indexed height", func(t *testing.T) {
+		t.Parallel()
 
-	require.NotNil(t, header)
-	assert.Equal(t, dps.FlowMainnet, header.ChainID)
+		// Use an empty DB without any entries.
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
 
-	_, err = c.Header(math.MaxUint64)
-	assert.Error(t, err)
+		c := chain.FromDisk(db)
+
+		_, err := c.Header(mocks.GenericHeight)
+		assert.Error(t, err)
+	})
 }
 
 func TestDisk_Commit(t *testing.T) {
-	db := populateDB(t)
-	defer db.Close()
-	c := chain.FromDisk(db)
+	t.Run("nominal case", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		require.NoError(t, db.Update(operation.IndexBlockHeight(mocks.GenericHeight, mocks.GenericHeader.ID())))
+		require.NoError(t, db.Update(operation.IndexStateCommitment(mocks.GenericHeader.ID(), mocks.GenericCommit(0))))
```

Comment:
Hmmmm. I'm not sure I like nesting it like this. I'm really a sucker for the repetitive style:

```go
err := db.Update(...)
require.NoError(t, err)
```

Just more readable, imo.

Reply:
I'm fine with it if it's your personal preference, but I don't feel like it improves readability at all for me :p The `db.Update(operation.XXX` is very early in the line, so it's pretty clear what each does, and it's neater than having many cases of error definition, having to make sure that the first one has a `:=` and not the following ones, not to mention it breaks when we want to reorder the DB operations (because of the `:=`).

If you insist on it though, I'm fine with changing it, as those are minor gripes I have with it overall, and it's not the end of the world for me.